### PR TITLE
Tools - HEMTT preprocess is now an object

### PR DESCRIPTION
**Pull request by BrettMayson**
**When merged this pull request will:**
- Update `preprocess` in addon.toml to be an object (HEMTT 1.7.1)

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: tools/make.py [hunk 1]
```diff
@@ -1302,13 +1302,13 @@ def main(argv):
                     addonTomlPath = os.path.join(work_drive, prefix, module, "addon.toml")
                     if os.path.isfile(addonTomlPath):
                         with open(addonTomlPath, "r") as f:
-                            skipPreprocessing = "preprocess = false" in f.read() #python 3.11 has real toml but this is fine for now
+                            skipPreprocessing = "[preprocess]\nenabled = false" in f.read() #python 3.11 has real toml but this is fine for now
```
Comment (PabstMirror) [inline tools/make.py:1305]:
```suggestion
                            skipPreprocessing = "[preprocess]\nenabled = false" in f.read() or "preprocess = false" in f.read()
                            #python 3.11 has real toml but this is fine for now
```

File: tools/sqfvmChecker.py [hunk 2]
```diff
@@ -29,11 +29,11 @@ def get_files_to_process(basePath):
                 addonTomlPath = os.path.join(root, "addon.toml")
                 if os.path.isfile(addonTomlPath):
                     with open(addonTomlPath, "r") as f:
-                        skipPreprocessing = "preprocess = false" in f.read()
+                        skipPreprocessing = "[preprocess]\nenabled = false" in f.read()
```
Comment (PabstMirror) [inline tools/sqfvmChecker.py:32]:
```suggestion
                        skipPreprocessing = "[preprocess]\nenabled = false" in f.read() or "preprocess = false" in f.read()
```

File: tools/sqfvmChecker.py [hunk 3]
```diff
@@ -29,11 +29,11 @@ def get_files_to_process(basePath):
...
                 addonTomlPath = os.path.join(os.path.dirname(root), "addon.toml")
                 if os.path.isfile(addonTomlPath):
                     with open(addonTomlPath, "r") as f:
-                        skipPreprocessing = "preprocess = false" in f.read()
+                        skipPreprocessing = "[preprocess]\nenabled = false" in f.read()
```
Comment (PabstMirror) [inline tools/sqfvmChecker.py:36]:
```suggestion
                        skipPreprocessing = "[preprocess]\nenabled = false" in f.read() or "preprocess = false" in f.read()
```

File: tools/make.py (hunk 1 above)
Comment (BrettMayson) [reply tools/make.py:1305]: What is the purpose of including this if it won't be used any more?

Comment (PabstMirror) [reply tools/make.py:1305]: I'd preferer to keep backward compatibility
I use these tools for other projects

Comment (BrettMayson) [reply tools/make.py:1305]: This is being deprecated in HEMTT as it was an outlier in the way it was formatted, it'll eventually be removed, you should update your other projects

**Review by jonpas — APPROVED** [review]

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**